### PR TITLE
PLAY-590 Player should stop playing when tab is not visible

### PR DESCRIPTION
**Pull request by mcm69**

### CODE REVIEW

**Comment by mcm69** [comment]
@shamansir plz review

File: src/anm/player_manager.js [hunk 1]
```diff
@@ -40,4 +45,23 @@ PlayerManager.prototype.getPlayer = function(cvs_id) {
...
+            player.pause();
+        } else if (!hidden && player._pausedViaHidden) {
+            player._pausedViaHidden = false;
+            player.play(player.state.from);
+        }
+    }
+};
+
+var manager = new PlayerManager();
+engine.onDocumentHiddenChange = function(hidden) {
```
Comment (shamansir) [inline src/anm/player_manager.js]: I think modifying something in the engine from outside is not a very good way, could it be refactored like this?:

`engine.onDocumentHiddenChange(function(hidden) { manager.handleDocumentHiddenChange(hidden); });`, which in its turn calls `addEventListener` with corresponding event, if it's there, or does nothing if it is not supported?

Comment (mcm69) [inline src/anm/player_manager.js]: sounds good to me

File: src/anm/player_manager.js [hunk 2]
```diff
@@ -40,4 +45,23 @@ PlayerManager.prototype.getPlayer = function(cvs_id) {
     return this.hash[cvs_id];
 };
 
-module.exports = new PlayerManager();
+PlayerManager.prototype.handleDocumentHiddenChange = function(hidden) {
```
Comment (shamansir) [inline src/anm/player_manager.js]: please add docs that this method is private.

File: src/engine/dom-engine.js [hunk 3]
```diff
@@ -981,5 +981,34 @@ $DE.isLocal = local;
...
+} else if (typeof document.mozHidden !== "undefined") {
+  hidden = "mozHidden";
+  visibilityChange = "mozvisibilitychange";
+} else if (typeof document.msHidden !== "undefined") {
+  hidden = "msHidden";
+  visibilityChange = "msvisibilitychange";
+} else if (typeof document.webkitHidden !== "undefined") {
+  hidden = "webkitHidden";
+  visibilityChange = "webkitvisibilitychange";
+}
```
Comment (RobertoUa) [inline src/engine/dom-engine.js]: so much code duplicated. Why not just prepend ['webkit', 'ms', 'moz'] to 'Hidden' and 'visibilitychange'.
Sorry, I'm just DRY fanboy.

**Pull request merged**